### PR TITLE
ensure activity list translations update after language change

### LINKED ISSUE #231: Caching on account activity
On TestFlight version 87, switch the app from Chinese to English. the account activity page shows cached strings.

IMG_E887FC09708C-1

### PR DESCRIPTION
**Pull request by tyler-whitman**
closes #231

### CODE REVIEW

File: src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
+      t('transactions.filter', { returnObjects: true }) as Record<
+        string,
+        string
+      >,
+    [t],
+  )
   const dispatch = useAppDispatch()
-  const [data] = useState(
-    FilterKeys.map((value) => ({ label: filters[value], value })),
+  const data = useMemo(
```
Comment (matthewcarlreetz) [inline src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx:23]: nice!

File: src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
+    [filters],
   )
 
   const onFilterChanged = useCallback(
     (_itemValue, itemIndex) => {
       const nextFilter = FilterKeys[itemIndex]
-      animateTransition()
-      dispatch(accountSlice.actions.resetActivityChart())
-      dispatch(activitySlice.actions.setFilter(nextFilter))
+      if (nextFilter !== filter) {
```
Comment (matthewcarlreetz) [inline src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx:31]: I'm curious if this was related to the language change or just a general optimization?

File: src/features/wallet/root/ActivityCard/ActivityItem.tsx [hunk 3]
```diff
@@ -88,11 +88,12 @@ const ActivityItem = ({
...
+  ActivityItem,
+  (prev, next) =>
     prev.hash === next.hash &&
     prev.isFirst === next.isFirst &&
     prev.isLast === next.isLast &&
-    prev.time === next.time
-  return areEqual
-})
+    prev.time === next.time &&
+    prev.title === next.title,
```
Comment (matthewcarlreetz) [inline src/features/wallet/root/ActivityCard/ActivityItem.tsx:98]: nice.

**Review by matthewcarlreetz — APPROVED** [review]
Seems to work well 👍

File: src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx:31]: Just a general optimization, but when you change the language this gets triggered a couple times so figured it was worth it to check

Comment (tyler-whitman) [reply src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx:31]: The animation I removed does seem to cause bugs though, I sometimes see a overlap of the graphs etc so I removed it for now

Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx:31]: ok, sounds good.

**Pull request merged**